### PR TITLE
Use FQCN injected for factory services (#47)

### LINKED ISSUE #47: Improve plugin's factories
The FQCN of the class to instantiate should be passed from dependency injection.

### PR DESCRIPTION
**Pull request by lruozzi9**
Fixes #47 

Changes proposed in this pull request:
- Use FQCN injected for factory services

### CODE REVIEW

File: src/Factory/ActiveCampaign/ConnectionFactory.php [hunk 1]
```diff
@@ -4,17 +4,24 @@
 
 namespace Webgriffe\SyliusActiveCampaignPlugin\Factory\ActiveCampaign;
 
-use Webgriffe\SyliusActiveCampaignPlugin\Model\ActiveCampaign\Connection;
 use Webgriffe\SyliusActiveCampaignPlugin\Model\ActiveCampaign\ConnectionInterface;
 
 final class ConnectionFactory implements ConnectionFactoryInterface
 {
+    public function __construct(
+        private string $connectionFQCN
```
Comment (LucaGallinari) [inline src/Factory/ActiveCampaign/ConnectionFactory.php:12]: We could have used a generic "classFQCN" so it's easier to maintain among all Factories, you can even put it in an asbtract class or a trait and use this in all of them

@lruozzi9 @mmenozzi what do you think?

**Review by LucaGallinari — CHANGES_REQUESTED** [review]

File: src/Factory/ActiveCampaign/ConnectionFactory.php (hunk 1 above)
Comment (lruozzi9) [reply src/Factory/ActiveCampaign/ConnectionFactory.php:12]: Ok, but how can you handle all the arguments of the create method?

Comment (LucaGallinari) [reply src/Factory/ActiveCampaign/ConnectionFactory.php]: Everything else remain the same. Take a look at my commit

Comment (lruozzi9) [reply src/Factory/ActiveCampaign/ConnectionFactory.php]: What I had in mind was to make a single factory class for all models. However, it becomes difficult to type the parameters and then do all the tests on these factories ... For the moment let's keep it that way!

**Pull request merged**